domain/provider.go

```diff
@@ -26,6 +27,7 @@ type Role struct {
 	Name        string        `json:"name" yaml:"name" validate:"required"`
 	Description string        `json:"description,omitempty" yaml:"description"`
 	Permissions []interface{} `json:"permissions" yaml:"permissions" validate:"required"`
+	Type        string        `json:"type,omitempty" yaml:"type"` // not required to support backward compatible to other provider
```

Comment:
this change is no longer required, let's remove it from here

Reply:
Will it broken when we remove it? since latest protobuff contain these field.
- https://github.com/goto/proton/commit/5709addd6a1a35b6c085eca96ae2c1deaf3842cf

Reply:
Resolved: https://github.com/goto/guardian/pull/190/commits/6bd8fdffe0d3723f936bc84ddaf326d111b06d9b